### PR TITLE
added threading section

### PR DESCRIPTION
**Pull request by Omarelsegeiny**
Should be added as a subsection into the technical low-level overview

### CODE REVIEW

File: docs/build/html/_sources/threads.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+- ``blip_image_thread.py``
+    Class that represents a thread instance salesforce BLIP image captioning model. Includes
+    a basic constructor, along with setters and getters for properties needed to retun for the output
+    as shown in the ``image_processer.py`` file.
+
+    Contains a ``run`` function: returns the result of the BLIP image captioning model, which is needed
+    for the CSV output.
+- ``easy_ocr_thread.py``
+    Class that represents a thread instance easyOCR model. Includes
+    a basic constructor, along with setters and getters for properties needed to retun for the output
```
Comment (KingPhilip14) [inline docs/build/html/_sources/threads.rst:21]: Misspelled "return" while saying, "...properties needed to **_retun_** for the output..."

File: docs/build/html/_sources/threads.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+We have implemented multithreading into the project for effeciency.
+The way we have done this is by creating classes of the two models used
+in this project, so they many return an instance of a thread to be used in the
+``image_processer.py`` file.
+
+    The Cohere LLM does not have a multihreading implementation, as it needs to wait for the input
+    from both the easyOCR and blip-image models.
+
+- ``blip_image_thread.py``
+    Class that represents a thread instance salesforce BLIP image captioning model. Includes
```
Comment (KingPhilip14) [inline docs/build/html/_sources/threads.rst:13]: The writing you've done overall is well done. I would recommend taking the docstrings from each method in these classes and expanding on them here. That is, take the docstrings that explains what each method is and expand on them as you see fit. If the docstrings are written well enough, they can be put here as is.

File: docs/build/html/_sources/threads.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    a basic constructor, along with setters and getters for properties needed to retun for the output
+    as shown in the ``image_processer.py`` file.
+
+    Contains a ``run`` function: returns the result of the BLIP image captioning model, which is needed
+    for the CSV output.
+- ``easy_ocr_thread.py``
+    Class that represents a thread instance easyOCR model. Includes
```
Comment (KingPhilip14) [inline docs/build/html/_sources/threads.rst:20]: Same thing here, where any docstrings in this file should be added here and expanded upon as needed.

**Review by KingPhilip14 — CHANGES_REQUESTED** [review]
If we can add the docstrings to the rst file, I think that will help since we're trying to explain the code more in depth. Explaining each method would be best to accomplish this.

File: docs/source/threads.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        # instantiating captions container for conditonal and unconditional outputs
+        self.captions: tuple[str, str] = ('', '')
+
+        self.image: Image = img
+
+        # processer and model instantiation from blip_image.py class
+        self.processor = BlipProcessor.from_pretrained("Salesforce/blip-image-captioning-large")
+        self.model = BlipForConditionalGeneration.from_pretrained("Salesforce/blip-image-captioning-large")
+
+    along with getters for properties needed to return for the output
```
Comment (KingPhilip14) [inline docs/source/threads.rst:31]: Capitalize "along"

File: docs/source/threads.rst [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    We do not have any getters/setters for the model and processer properties, as we will never need
+    to return instances of them not change them in any way.
+
+    Contains a ``run`` function that returns the result of the BLIP image captioning model, which is needed
+    for the CSV output:
+
+    .. code-block:: console
+
+        def run(self):
+        self.captions = bi.caption_image(self.image, self.processor, self.model)
```
Comment (KingPhilip14) [inline docs/source/threads.rst:61]: Make sure this line is indented where it says "self.captions"

**Review by KingPhilip14 — CHANGES_REQUESTED** [review]
Some small changes, but this is really good!

**Review by minisodan — APPROVED** [review]
be more specific on why we needed threading in the first paragraph

**Pull request merged**